pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -615,7 +616,17 @@ def part_end_event(next_part: ModelResponsePart | None = None) -> PartEndEvent |
                         next_part_kind=next_part.part_kind if next_part else None,
                     )
 
-                async for event in iterator:
+                async def chain_async_and_sync_iters(
+                    iter1: AsyncIterator[ModelResponseStreamEvent], iter2: Iterator[ModelResponseStreamEvent]
+                ) -> AsyncIterator[ModelResponseStreamEvent]:
+                    async for event in iter1:
+                        yield event
+                    for event in iter2:  # pragma: no cover - idk why this isn't covered
```

Comment:
we'll definitely want to make sure this is covered :)

Reply:
this line is definitely running, the tests include `final_flush()` behavior and `final_flush()` in only called here, I just don't know why coverage isn't picking it up... the current CI coverage failure seems to agree that it's being covered but if I remove it and coverage fails in the next run I don't know what I'll do 🫣